docusaurus/docs/dev-docs/upgrade-tool.md

```diff
@@ -0,0 +1,164 @@
+
+The upgrade tool is a Strapi package and can be run from the CLI.
+
+## Version types
+
+Strapi version numbers respect the [semantic versioning](https://semver.org/) conventions:
+
+<ThemedImage
+  alt="Version numbers explained"
+  sources={{
+    light: '/img/assets/update-migration/version-numbers.png',
+    dark: '/img/assets/update-migration/version-numbers_DARK.png',
+  }}
+/>
+
+- The first number is the **major** version number.
+- The second number is the **minor** version number.
+- The third number is the **patch** version number.
+
+The upgrade tool allows upgrading to a major, minor, or patch version.
+
+| The latest Strapi v4 version is v4.16.2 and my Strapi application is currently on… | If I run… | My Strapi application will be upgraded to … |
+|----|----|----|
+| v4.14.1 | `npx @strapi/upgrade patch` | v4.14.6<br/><br/>(because v4.14.6 is the latest patch version for the v4.14 minor version) |
+| v4.14.1 | `npx @strapi/upgrade minor` | v4.16.2 |
+| v4.14.1 | `npx @strapi/upgrade major` | Nothing.<br/><br/>I first need to run `npx @strapi/upgrade minor` to upgrade to v4.16.2. |
+| v4.16.2 | `npx @strapi/upgrade major` | v5.0.0  |
+
+## Upgrade to a major version
+
+Run the upgrade tool with the `major` parameter to upgrade the project to the latest major version of Strapi:
+
+```bash
+npx @strapi/upgrade major
+```
+
+During the upgrade process, the application dependencies are updated and installed, and the codemods up until the targeted major version are executed.
+
+:::note
+If your application is not already running the latest minor and patch version in the current major, the `major` upgrade is prevented, and you will first need to upgrade to the latest [minor.patch](#upgrade-to-a-minor-version) version in the current major version. This means that moving from v4.14.4 to v5.0.0 is a 2-step process because the latest v4 version is v4.16.2.
+:::
+
+## Upgrade to a minor version
+
+Run the upgrade tool with the `minor` parameter to upgrade the project to the latest minor and patch version of Strapi:
+
+```bash
+npx @strapi/upgrade minor
+```
+
+During the upgrade process, the project dependencies are updated and installed, and the codemods up until the targeted patch are executed (if any).
+
+## Upgrade to a patch version
+
+Run the upgrade tool with the `patch` parameter to upgrade the project to the latest major version of Strapi:
+
+```bash
+npx @strapi/upgrade patch
+```
+
+During the upgrade process, the project dependencies are updated and installed, and the codemods up until the targeted patch are executed (if any).
+
+## Run codemods only
```

Comment:
This is still TBD for now.
This **should** be in the initial release but maybe we can add this later or add a "coming later/soon"?
What do you think?

Reply:
That's a good point. I would comment out these parts so they don't appear on the website, and whenever the feature is ready we would uncomment it. What do you think?

(If we're 100% sure it's going to be part of the feature for the stable v5 release, I could also add a callout mentioning something about codemods only is coming soon — I just don't want to disappoint users if in the end it's decided to be out of scope 😅)

Reply:
Let's comment it out & add it back when the command will be there :slightly_smiling_face: